smallstep/cli
# Add Winget automation to GoReleaser

~~This is blocked on https://github.com/goreleaser/goreleaser/pull/4196~~

To test the Winget package itself:
- Create a prerelease on this branch
- Look in the `dist` folder in the workflow artifacts, for the Winget manifest files
- Compare them to my [hand-crafted winget yaml files for `step`](https://github.com/microsoft/winget-pkgs/tree/master/manifests/s/Smallstep/step/0.24.1), make sure all the critical fields are present and populated properly
- Create a PR on [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) using the yaml files, and see if it passes validations

To test the Winget PR automation:
- [x] Create a smallstep/winget-pkgs repo that is a fork of [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs).
- [x] In .goreleaser.yml, set `skip_upload: false` and then push a pre-release
- [x] Make sure the generated YAML looks great.
- [x] See if a PR is created properly in smallstep/winget-pkgs.

Before merging this:
- Configure `.goreleaser.yml` to use `microsoft/winget-pkgs` as the base package (uncomment lines 351-354)
- Change `skip_upload` to `auto`, so that prereleases no longer trigger PRs
